modules/enterprise/server/jar/src/main/java/org/rhq/enterprise/server/resource/ResourceManagerBean.java

```diff
@@ -783,6 +789,82 @@ private boolean uninventoryResourceBulkDeleteAsyncWork(Subject overlord, int res
                 resourceIds);
         }
 
+        // If this resource had packages installed, remove their version and bits
+        if (installedPackages.size() > 0) {
+            hasErrors |= cleanOrphanedPackages(overlord, installedPackages);
+        }
+
+        return hasErrors;
+    }
+
+    @SuppressWarnings("unchecked")
+    private List<Integer> getInstalledPackages(Integer resourceId) {
+
+        List<Integer> installed_packages = new ArrayList<Integer>();
```

Comment:
better to stick with camelCase, I think.   And again, installedPackageIds would be more clear.